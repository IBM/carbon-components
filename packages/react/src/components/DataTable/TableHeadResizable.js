/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import PropTypes from 'prop-types';
import React from 'react';
import { useResizableColumnsTableHead } from './tools/columnResize';

const TableHeadResizable = ({ children, isResizable, ...rest }) => {
  const { ref } = useResizableColumnsTableHead();

  if (isResizable) {
    return (
      <thead ref={ref} {...rest}>
        {children}
      </thead>
    );
  }

  return <thead {...rest}>{children}</thead>;
};

TableHeadResizable.propTypes = {
  /**
   * Pass in children that will be embedded in the table header label
   */
  children: PropTypes.node,

  /**
   * `false` If true, will apply resizable styles
   */
  isResizable: PropTypes.bool,
};

TableHeadResizable.displayName = 'TableHeadResizable';

export default TableHeadResizable;
