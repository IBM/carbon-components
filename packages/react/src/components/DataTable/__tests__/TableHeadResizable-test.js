/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Table } from '..';
import TableHeadResizable from '../TableHeadResizable';

describe('DataTable.TableHeadResizable', () => {
  it('should render', () => {
    const wrapper = mount(
      <Table>
        <TableHeadResizable className="custom-class" />
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with isResizable', () => {
    const wrapper = mount(
      <Table>
        <TableHeadResizable isResizable className="custom-class" />
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
