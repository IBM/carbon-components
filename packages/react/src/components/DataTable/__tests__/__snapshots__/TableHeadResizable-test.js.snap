// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableHeadResizable should render 1`] = `
<Table
  isSortable={false}
>
  <table
    className="bx--data-table bx--data-table--no-border"
  >
    <TableHeadResizable
      className="custom-class"
    >
      <thead
        className="custom-class"
      />
    </TableHeadResizable>
  </table>
</Table>
`;

exports[`DataTable.TableHeadResizable should render with isResizable 1`] = `
<Table
  isSortable={false}
>
  <table
    className="bx--data-table bx--data-table--no-border"
  >
    <TableHeadResizable
      className="custom-class"
      isResizable={true}
    >
      <thead
        className="custom-class"
      />
    </TableHeadResizable>
  </table>
</Table>
`;
